cmd/packages.pony

```diff
@@ -110,11 +119,18 @@ class val Package is Comparable[Package box]
   fun string(): String iso^ =>
     "-".join([name; channel; version; platform()].values())
 
-type CPU is AMD64
-primitive AMD64
+type CPU is (AMD64 & _CPU)
+interface val _CPU is (Equatable[_OS] & Stringable)
+primitive AMD64 is _OS
+  fun string(): String iso^ => "x86_64".string()
 
-type OS is (Linux | Darwin)
-primitive Linux
-primitive Darwin
+type OS is ((Linux | Darwin | FreeBSD) & _OS)
+interface val _OS is (Equatable[_OS] & Stringable)
+primitive Linux is _OS
```

Comment:
I think I just found a reason to want to have default implementation on interfaces.